kornia/augmentation/random_generator/_2d/random_rain.py

```diff
@@ -74,3 +77,60 @@ def forward(self, batch_shape: tuple[int, ...], same_on_batch: bool = False) ->
             'drop_height_factor': drop_height_factor,
             'drop_width_factor': drop_width_factor,
         }
+
+    def random_rain_augmentation(
+        self,
+        image: torch.Tensor,
+        num_raindrops: int = 100,
+        min_length: int = 5,
+        max_length: int = 15,
+        color: torch.Tensor = torch.tensor([255]),
+    ) -> torch.Tensor:
+        """Apply random rain augmentation to the input image.
+        Args:
+            image (torch.Tensor): the input image with shape :math:`(C,H,W)`.
+            num_raindrops (int, optional): number of raindrops to draw. Defaults to 100.
+            min_length (int, optional): minimum length of a raindrop. Defaults to 5.
+            max_length (int, optional): maximum length of a raindrop. Defaults to 15.
+            color (torch.Tensor, optional): the color of the raindrops. Defaults to torch.tensor([255]).
+
+        Returns:
+            torch.Tensor: The image with raindrops.
+        Doctests:
+        >>> import torch
+        >>> aug = RainGenerator(100, 5, 15)
+        >>> image = torch.zeros(3, 10, 10)
+        >>> no_rain = aug.random_rain_augmentation(image, 0)
+        >>> torch.equal(no_rain, image)
+        True
+        >>> rain_img = aug.random_rain_augmentation(image, 1000)
+        >>> torch.equal(rain_img, image)
+        False
+        >>> custom_color = torch.tensor([50, 50, 50])
+        >>> colored_rain_img = aug.random_rain_augmentation(image, 1000, color=custom_color)
+        >>> torch.any(colored_rain_img == 50)
+        True
+        """
+        H, W = image.shape[1], image.shape[2]
+        for _ in range(num_raindrops):
+            # Randomly select the starting point
+            start_x = random.randint(0, W - 1)
```

Comment:
Can we compute these random parameters within some of the base aug methods? @shijianjian